ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -389,6 +390,47 @@ def get_provider_slot(self, parent: Operator) -> OutputSlot:
         return opReader.Output
 
 
+class DummyDatasetInfo(DatasetInfo):
+    """Special DatasetInfo for datasets that can't be found in headless mode
+    """
+
+    def __init__(self, *, project_file: h5py.File = None, **info_kwargs):
+        super().__init__(**info_kwargs)
+
+    @property
+    def legacy_location(self) -> str:
+        return "Outer Space"
+
+    @property
+    def effective_path(self) -> str:
+        return "Outer Space"
+
+    @property
+    def display_string(self) -> str:
+        return "Dummy Zero Array"
+
+    def to_json_data(self) -> Dict:
+        raise NotImplemented("Dummy Slots should not be serialized!")
+
+    def get_provider_slot(self, parent: Operator) -> OutputSlot:
+        opZero = OpZeroSource(shape=self.laneShape, dtype=self.laneDtype, axistags=self.axistags, parent=parent)
+        return opZero.Output
+
+    @classmethod
+    def from_h5_group(cls, data: h5py.Group, params: Dict = None):
```

Comment:
The reason the `params` dict is accepted in the base `DatasetInfo.from_h5_group` is just so that the child classes can parse their specific fields and pass those up those up to the super method (which i'm not sure was good design anyway). If you think it might be useful to override some of the parameters, like `laneShape` or `default_tags`, then maybe it would be a good idea to have those as explicit parameters of the `from_h5_group` method

Reply:
do you imply I am misusing it? I see no other way to pass up the missing values that have not been serialized, but are needed for initialization...

Reply:
Ah, now I see what's happening. I myself added the `params` parameter to all of the `from_h5_group` methods, which now I think was a bad idea, because no client of those methods ever uses that parameter, and probably never should; it's a dictionary, which can take anything and ignore anything, so it would be very non-intuitive for the clients of those methods and is only justifiable in the base class, since there it is needed to support the specialized `DatasetInfo`s. If we actually wanted the specialized DatasetInfos to allow for some parameters to be overridden, those should be explicit parameters in their respective `from_h5_group` method. 

Ideally the `from_h5_group` methods should be `@staticmethod`s rater than `@classmethod`s, and the base `from_h5_group` should be renamed but still be a `@classmethod` to support the specialized DatasetInfo classes.

Anyway, I guess we can ignore this for now. I can get back to it later and then fix all of the specialized DatasetInfos.

Reply:
:+1: